### ISSUE TITLE
Test results plugin

### ISSUE THREAD

**Issue by Tlacenka** [code-pushup/cli#477]
As a user, I want to be able to integrate my test suites into Code PushUp and aggregate the results in my report.

## Research
- [ ] Research has been done on test results formats that need to be supported
- [ ] Design additional `details` properties required for this plugin to contain all necessary information
   - [ ] (can be a separate issue) Research has been done on how to set up a blob storage for videos or screenshots.
   - [ ] A better way of displaying an error message than just raw text with full call stack.

## Acceptance criteria
- [ ] There is one plugin that aggregates all tools and their test results.
- [ ] Decide on the approach:
   - Either there is a group for every testing tool (which may offer different types of testing). For example Cypress group would contain component and E2E tests. Jest group would contain unit and integration tests.
   - Or there is one generic audit with configuration for every tool (but then we would still need to somehow insert the icon, documentation and descriptions that are relevant only for that tool and have a set of supported ones).
- [ ] An audit would be a specific test result from a given tool (successor of test results format spike).
- [ ] An issue would be a failing test with information about the test (name, file, error message, screenshot?).
- [ ] The score and value would refer to the amount of passing tests (range 0-1 or 0-100 respectively).

**Comment by Tlacenka** [comment]
# Test results format

**TL;DR: JUnit** is the most commonly used (natively supported by all researched test frameworks). Screenshots may be attached to a test result inside [`attachments`](https://www.npmjs.com/package/mocha-junit-reporter#attachments).

## Common test results formats
- [xUnit](https://xunit.net/): XML report made primarily for .NET framework.
- [NUnit](https://docs.nunit.org/articles/nunit/intro.html): XML report, easier for referencing attachments in Azure DevOps.
- [JUnit](https://github.com/testmoapp/junitxml): XML report made primarily for Java unit test framework. Natively supported in most tools.
- JSON and HTML are also common formats for test result reports.

## Compatibility table
| Framework | JUnit | xUnit | NUnit | JSON | HTML | Custom reporter support |
| --- | --- | --- | --- | --- | --- | --- |
| [Cypress](https://docs.cypress.io/guides/tooling/reporters) | **yes (native)** | - | - | - | - | yes (mocha reporters) |
| [Playwright](https://playwright.dev/docs/test-reporters) | **yes (native)** | - | - | yes (native) | yes (native) | yes (reporters) |
| [Jest](https://jestjs.io/docs/cli#--reporters) | **yes (native)** | yes ([jest-xunit](https://www.npmjs.com/package/jest-xunit) package) | yes ([jest-nunit-reporter](https://www.npmjs.com/package/jest-nunit-reporter) package) | yes ([jest-json-reporter](https://www.npmjs.com/package/jest-json-reporter) package) | yes ([jest-html-reporter](https://www.npmjs.com/package/jest-html-reporter) package) | yes (npm packages) |
| [Pytest](https://docs.pytest.org/en/7.1.x/how-to/output.html) | **yes (native)** | - | yes ([pytest-nunit](https://pypi.org/project/pytest-nunit/) plugin) | yes ([pytest-json](https://pypi.org/project/pytest-json/) plugin) | yes ([pytest-html](https://github.com/pytest-dev/pytest-html) plugin) | yes (plugins) |
| [.NET](https://github.com/microsoft/vstest/blob/main/docs/report.md#available-test-loggers) | **yes (native)** | yes (native) | yes (native) | yes (native) | yes (native) | yes (test loggers) |
